// pushpie-logger/index.js
var log4js = require('log4js');
log4js.configure({
    appenders: [
        {
            type: 'console'
        },
        {
            type: 'dateFile',
            filename: 'logs/log',
            pattern: "_yyyy-MM-dd",
            maxLogSize: 1024,
            alwaysIncludePattern: false,
            backups: 4,
            category: [ 'normal','console' ]
        },
        {
            type: 'dateFile',
            filename: 'logs/err',
            pattern: "_yyyy-MM-dd",
            maxLogSize: 1024,
            alwaysIncludePattern: false,
            backups: 4,
            category: 'error'
        }
    ],
    replaceConsole: true
});
var normalLogger = log4js.getLogger('normal');
normalLogger.setLevel('INFO');

var errLogger = log4js.getLogger('error');
errLogger.setLevel('WARN');

//err log文件包含fatal、error、warn这三种级别
//normal log文件只有info
module.exports = {
	'log4js': log4js,
	'normalLogger': normalLogger,
	'errLogger': errLogger,
	'write': function(type, msg){
		if(type == 'fatal'){
			errLogger.fatal(msg);
		}else if(type == 'error'){
			errLogger.error(msg);
		}else if(type == 'warn'){
			errLogger.warn(msg);
		}else{ // info
			normalLogger.info(msg);
		}
	}
};
